File: experiments/runs/2026-08-10_15-31-22_claim-office-example-mapping_v3-basic-tdd_opus-5-no-thinking/src/priceList.ts
import { ClaimOfficeError } from './types.js';

interface Price {
  insuranceValue: number;
  basePremium: number;
}

const MAIN_ITEMS: Record<string, Price> = {
  sword: { insuranceValue: 1000, basePremium: 100 },
  amulet: { insuranceValue: 600, basePremium: 60 },
  staff: { insuranceValue: 800, basePremium: 80 },
  potion: { insuranceValue: 400, basePremium: 40 },
};

const COMPONENTS = ['rune', 'moonstone'];

export const COMPONENT_PRICE: Price = { insuranceValue: 250, basePremium: 25 };

/** Special base premium for a building block of exactly 3 alike components. */
export const COMPONENT_BLOCK_PREMIUM = 60;
export const COMPONENT_BLOCK_SIZE = 3;

export const isComponent = (type: string): boolean => COMPONENTS.includes(type);

export function priceOf(type: string): Price {
  if (isComponent(type)) return COMPONENT_PRICE;

  const price = MAIN_ITEMS[type];
  if (!price) throw new ClaimOfficeError(`unknown item type: ${type}`);
  return price;
}

File: experiments/runs/2026-09-22_00-18-38_claim-office-python-example-mapping_baseline-inline-tdd-v1-cc_opus-5-no-thinking/src/pricelist.py
"""The MHPCO price list: insurance values and base premiums per item type."""

from fractions import Fraction

MAIN_ITEMS = {
    "sword": (1000, 100),
    "amulet": (600, 60),
    "staff": (800, 80),
    "potion": (400, 40),
}

COMPONENT_TYPES = ("rune", "moonstone")
COMPONENT_VALUE = 250
COMPONENT_PREMIUM = 25
BLOCK_SIZE = 3
BLOCK_PREMIUM = 60


class UnknownItemType(Exception):
    """Raised when an item type is not in the MHPCO price list."""


def is_component(item_type):
    return item_type in COMPONENT_TYPES


def insurance_value(item_type):
    if item_type in MAIN_ITEMS:
        return MAIN_ITEMS[item_type][0]
    if is_component(item_type):
        return COMPONENT_VALUE
    raise UnknownItemType(f"unknown item type: {item_type}")


def base_premium(item_type):
    if item_type in MAIN_ITEMS:
        return Fraction(MAIN_ITEMS[item_type][1])
    if is_component(item_type):
        return Fraction(COMPONENT_PREMIUM)
    raise UnknownItemType(f"unknown item type: {item_type}")


def block_premium(item_type, count):
    """Base premium for `count` alike components of one type.

    The MHPCO grants its block rate only for exactly three alike components;
    any other count is billed per component.
    """
    if count == BLOCK_SIZE:
        return Fraction(BLOCK_PREMIUM)
    return base_premium(item_type) * count
